internal/services/arckubernetes/arc_kubernetes_cluster_resource.go

```diff
@@ -203,12 +309,45 @@ func resourceArcKubernetesClusterUpdate(d *pluginsdk.ResourceData, meta interfac
 		return err
 	}
 
-	props := arckubernetes.ConnectedClusterPatch{
-		Tags: tags.Expand(d.Get("tags").(map[string]interface{})),
+	resp, err := client.ConnectedClusterGet(ctx, *id)
+	if err != nil {
+		return fmt.Errorf("retrieving %s: %+v", id, err)
+	}
+
+	payload := resp.Model
+
+	if d.HasChange("aad_profile") {
+		payload.Properties.AadProfile = expandArcKubernetesClusterAadProfile(d.Get("aad_profile").([]interface{}))
+	}
+
+	if d.HasChange("tags") {
+		payload.Tags = tags.Expand(d.Get("tags").(map[string]interface{}))
+	}
+
+	if d.HasChange("azure_hybrid_benefit") {
+		payload.Properties.AzureHybridBenefit = pointer.To(arckubernetes.AzureHybridBenefit(d.Get("azure_hybrid_benefit").(string)))
 	}
 
-	if _, err := client.ConnectedClusterUpdate(ctx, *id, props); err != nil {
-		return fmt.Errorf("updating %s: %+v", *id, err)
+	if d.HasChange("arc_agent_desired_version") {
+		if desiredVersion := d.Get("arc_agent_desired_version").(string); desiredVersion != "" {
+			payload.Properties.ArcAgentProfile.DesiredAgentVersion = pointer.To(desiredVersion)
+		} else {
+			payload.Properties.ArcAgentProfile.DesiredAgentVersion = nil
+		}
+	}
+
+	if d.HasChange("arc_agent_auto_upgrade_enabled") {
+
+		autoUpgradeOption := arckubernetes.AutoUpgradeOptionsEnabled
+		if !d.Get("arc_agent_auto_upgrade_enabled").(bool) {
+			autoUpgradeOption = arckubernetes.AutoUpgradeOptionsDisabled
+		}
+
+		payload.Properties.ArcAgentProfile.AgentAutoUpgrade = pointer.To(autoUpgradeOption)
+	}
+
+	if err := client.ConnectedClusterCreateThenPoll(ctx, *id, *payload); err != nil {
```

Comment:
the function name should be `CreateOrUpdateThenPoll` ?

Reply:
There is no `CreateOrUpdateThenPoll` method, and the `UpdateThenPoll` method is a patch method, so only the `CreateThenPoll` method is useful here.